lib/browser/evaluate.js

```diff
@@ -56,3 +56,22 @@ exports.evaluate = function evaluate() {
       throw new Error(invocation);
   }
 };
+
+exports.evaluateAsync = function evaluateAsync() {
+  const args = _.toArray(arguments);
+  let clientFunction = args.pop();
+
+  const invocation =
+    'evaluateAsync(clientFunction) - requires (Function) clientFunction';
+  assert.truthy(invocation, clientFunction);
+
+  if (typeof clientFunction !== 'function') {
+    throw new Error(invocation);
+  }
+
+  clientFunction = `var done = arguments[arguments.length-1], args=[].slice.call(arguments, 0, arguments.length-1);
+      return new Promise(
+        (resolve) => resolve((${clientFunction}).apply(this, args))
+      ).then(done)`;
```

Comment:
The `.then(done)` means that rejections just make this hang (unless I'm missing something) - can we make sure that we have a test case with an error?

Reply:
good point - contrary to the docs, the `done()` doesn't accept error-first-arg :-(

any thoughts on a good way to get the errors out?

Reply:
We could wrap this in `{ reason, value }` and convert to rejection in the API method?